### ISSUE TITLE
Allow additional user-data on EKS nodes

### ISSUE THREAD

**Issue by berndtj** [weaveworks/eksctl#69]
It would be great to be able easily add to the node template user-data.  While one could get most of the same functionality with custom AMIs, then you have to maintain AMIs.  For instance, I hacked the nodegroup template to add support for insecure registries (deploy pods from a registry deployed within the cluster):

```
"echo {\\\"insecure-registries\\\": [\\\"$DNS_CLUSTER_IP/8\\\"]} > /etc/docker/daemon.json" , "\n",
"systemctl restart docker", "\n",
```

**Comment by errordeveloper** [comment]
I agree, in general we should provide customisation, but I believe this is a special case and we are already discussing this with @jstrachan (see https://github.com/jenkins-x/jx/issues/87).

**Comment by berndtj** [comment]
Awesome @errordeveloper! Let me know if I can help.  BTW, I was sitting next to you at the openfaas meetup last week.

**Comment by errordeveloper** [comment]
@berndtj please feel free to explore this. At the moment we have CloudFormation templates generated with go-bindata and checked-in. I suppose it's be easiest to implement this as parameter in CloudFomation – wdyt? I'm not entirely sure about enabling it by default right now.

**Comment by archisgore** [comment]
+1 to this. A trivial use-case here is installing `amazon-efs-utils` on top of the latest AMI. Cutting a custom AMI seems like a LOT of work to have something as simple. I'll take a look at this.

**Comment by errordeveloper** [comment]
Archis, to me EFS seems like an add-on that we should enable with a flag -
would you agree?
To support it, we can discuss adding the user-space support upstream, and
of course installing it at boot wouldn't be a big deal either.

I am intending to make some changes to default user data, so I will
probably take a stab at parameterizing it while at it (see
https://github.com/weaveworks/eksctl/issues/104).

**Comment by errordeveloper** [comment]
This should get unblocked with #132.

**Comment by niranjan94** [comment]
@errordeveloper how can I add an additional command (or a set of commands) to be run at instance start for a windows instance part of a windows node group ? 

Some uses cases for this would be
- Setting the instance pagefile size
- Installing & configuring the cloudwatch agent for instance metrics collection.

If I try `preBootstrapCommands`, I get 

```bash
Error: preBootstrapCommands is not supported for WindowsServer2019CoreContainer nodegroups (path=nodeGroups[2].preBootstrapCommands)
```

**Comment by errordeveloper** [comment]
@niranjan94 please open another issue, that would be a feature request actually ;)

**Comment by niranjan94** [comment]
Thanks for your response @errordeveloper
Have opened an issue https://github.com/weaveworks/eksctl/issues/2003

Have also submitted a PR for the same.
